js/src/sdk/base.toolset.ts

```diff
@@ -219,23 +219,32 @@ export class ComposioToolSet {
         connectedAccountId?: string,
     ): Promise<Record<string, any>> {
         // Custom actions are always executed in the host/local environment for JS SDK
-        if(await this.isCustomAction(action)) {
+        if (await this.isCustomAction(action)) {
+            let accountId = connectedAccountId;
+            if (!connectedAccountId) {
+                // fetch connected account id
+                const connectedAccounts = await this.client.connectedAccounts.list({
+                    user_uuid: entityId
+                });
+                accountId = connectedAccounts.items[0].id;
+            }
+
             return this.customActionRegistry.executeAction(action, params, {
                 entityId: entityId,
                 connectionId: connectedAccountId
```

Comment:
The `connectionId` parameter in executeAction is still using `connectedAccountId` instead of the newly fetched `accountId`. This could cause the custom action to execute with undefined account ID.